Source/SteveSloth/Private/MyEnemyAIIdleState.cpp

```diff
@@ -0,0 +1,34 @@
+/****************************************************************************************
+* Copyright: SteveSloth
+ * Name: Aidan MacLeod
+ * Script: MyEnemyAIIdleState.cpp
+ * Date: April 29. 2024
+ * Description: Idle State Class for all enemies to inherit from
+ * TODO:
+ * Known Bugs:
+ ****************************************************************************************/
+
+#include "MyEnemyAIIdleState.h"
+
+
+
+
+
+MyEnemyAIIdleState::MyEnemyAIIdleState()
+{
+	IsIdle = false;
+	HasDetectedPlayer = false;
+}
+
+MyEnemyAIIdleState::~MyEnemyAIIdleState()
+{
+
+}
+
+void MyEnemyAIIdleState::DetectPlayer()
```

Comment:
1: This function is never called.
2: Should write the code to detect the player.
3: should setup a range to detect the player in, 100 units away 1000 units away. These things need to be used and adjusted.
4: If no player around for X amount of time go into patrol state
5: If player in range start to chase player.